config/config.py

```diff
@@ -336,211 +461,671 @@ def build_config(era, sample, channels, shifts):
+                ("et", "mt", "tt", "em", "ee", "mm"): {
+                    "apply_recoil_resolution_systematic": True,
+                    "apply_recoil_response_systematic": False,
+                    "recoil_systematic_shift_up": False,
+                    "recoil_systematic_shift_down": True,
+                },
+            },
+            producers={
+                ("et", "mt", "tt", "em", "ee", "mm"): met.ApplyRecoilCorrections
+            },
+        ),
+        samples=[
+            sample
+            for sample in available_sample_types
+            if sample not in ["data", "emb", "emb_mc"]
+        ],
     )
+    #########################
     # Jet energy resolution
-    shift_dict = {"JE_reso_shift": 1}
-    AddSystematicShift(
-        config, "jerUncUp", {"global": shift_dict}, [[JetEnergyCorrection, "global"]]
+    #########################
+    configuration.add_shift(
+        SystematicShift(
+            name="jerUncUp",
+            shift_config={
+                ("et", "mt", "tt", "em", "ee", "mm"): {"JE_reso_shift": 1},
+            },
+            producers={"global": jets.JetEnergyCorrection},
+        ),
+        samples=[
+            sample
+            for sample in available_sample_types
+            if sample not in ["data", "emb", "emb_mc"]
+        ],
     )
-    shift_dict = {"JE_reso_shift": -1}
-    AddSystematicShift(
-        config, "jerUncDown", {"global": shift_dict}, [[JetEnergyCorrection, "global"]]
+    configuration.add_shift(
+        SystematicShift(
+            name="jerUncDown",
+            shift_config={
+                ("et", "mt", "tt", "em", "ee", "mm"): {"JE_reso_shift": -1},
+            },
+            producers={"global": jets.JetEnergyCorrection},
+        ),
+        samples=[
+            sample
+            for sample in available_sample_types
+            if sample not in ["data", "emb", "emb_mc"]
+        ],
     )
+    #########################
     # Jet energy scale
-    # JEC_sources = '{"SinglePionECAL", "SinglePionHCAL", "AbsoluteMPFBias", "AbsoluteScale", "Fragmentation", "PileUpDataMC", "RelativeFSR", "PileUpPtRef"}'
-    # shift_dict = {"JE_scale_shift": 1, "JEC_shift_sources": JEC_sources}
-    # AddSystematicShift(config, "jecUncAbsoluteUp", {"global": shift_dict}, [[JetEnergyCorrection, "global"]])
-    # shift_dict = {"JE_scale_shift": -1, "JEC_shift_sources": JEC_sources}
-    # AddSystematicShift(config, "jecUncAbsoluteDown", {"global": shift_dict}, [[JetEnergyCorrection, "global"]])
+    #########################
+    JEC_sources = '{"SinglePionECAL", "SinglePionHCAL", "AbsoluteMPFBias", "AbsoluteScale", "Fragmentation", "PileUpDataMC", "RelativeFSR", "PileUpPtRef"}'
+    configuration.add_shift(
+        SystematicShift(
+            name="jecUncAbsoluteUp",
+            shift_config={
+                ("et", "mt", "tt", "em", "ee", "mm"): {
+                    "JE_scale_shift": 1,
+                    "JEC_shift_sources": JEC_sources,
```

Comment:
The `JE_scale_shift` and `JEC_shift_sources` are defined in the `"global"` scope. 
https://github.com/KIT-CMS/CROWN/blob/eff8756bcf070769078e2c7ed6d840eaac37f1e1/config/config.py#L117-L118 
Is there no problem to have them now twice in `"global"` and e.g. `"mt"` ?

Reply:
Yes, i adapted the code, so that it is possible to write just `"global"` here, now it looks much more reasonable